rpcs3/Emu/RSX/rsx_methods.cpp

```diff
@@ -1022,15 +1022,15 @@ namespace rsx
 
 			if (scale_y < 0 || scale_x < 0)
 			{
-				const auto packed_pitch = in_w * in_bpp;
+				const u32 packed_pitch = in_w * in_bpp;
 				temp1.resize(packed_pitch * in_h);
 
-				const s32 stride_y = (scale_y < 0 ? -1 : 1) * in_pitch;
+				const s64 stride_y = (scale_y < 0 ? -1 : 1) * (s64)in_pitch;
 
 				for (u32 y = 0; y < in_h; ++y)
 				{
 					u8 *dst = temp1.data() + (packed_pitch * y);
-					u8 *src = pixels_src + (y * stride_y);
+					u8 *src = pixels_src + ((s64)y * stride_y);
```

Comment:
why? is u32 * s64 really undefined? u32->s64 promotion should be safe, is it not?